app/src/main/java/com/github/sdp_begreen/begreen/firebase/FirebaseDB.kt

```diff
@@ -293,4 +301,40 @@ object FirebaseDB {
             null
         }
     }
+
+    /**
+     * Store the given [location] as the location of a recycling bin.
+     *
+     * @param location the location of the bin
+     *
+     * @return true if the location got stored in the database, and false if it failed
+     */
+    suspend fun storeBinLocation(location: LatLng): Boolean{
+        val freshId = databaseReference.child(BIN_LOCATION_PATH).push().key ?: return false
+        databaseReference.child(BIN_LOCATION_PATH).child(freshId).setValue(location).await()
+        return true
+    }
+
+    /**
+     * Retrieves the list of all bin locations currently present in the database
+     *
+     * @return the list of locations of all the bins, or null if an error happened
+     */
+    suspend fun getAllBinLocations(): List<LatLng>?{
+        // Looks like manually deleting the unchecked cast is the only way to remove the warnings
+        @Suppress("UNCHECKED_CAST")
+        val childrens = databaseReference.child(BIN_LOCATION_PATH).get().await().value
+                as? Map<String, Any> ?: return null
+
+        val locations = mutableListOf<LatLng>()
+        for (child in childrens.values) {
+            // Converts the Map back to a LatLng. If some child does not have the correct format, skip it
+            @Suppress("UNCHECKED_CAST")
+            val location = child as? HashMap<String, Double>? ?: child as? HashMap<String, Long>? ?: continue

```

Comment:
This way of iterating over element with the `continue` operator can be hard to understand. I recommend you to keep an easier code to understand like that :

```
for (child in childrens.values) {
    // Converts the Map back to a LatLng. If some child does not have the correct format, skip it
    @Suppress("UNCHECKED_CAST")
    val location = child as? HashMap<String, *>?
    if (location != null && location.containsKey("latitude") && location.containsKey("longitude")) {
        val lat = location["latitude"] as? Double
        val lng = location["longitude"] as? Double
        if (lat != null && lng != null) {
            locations.add(LatLng(lat, lng))
        }
    }
}

```

But feel free to use the one you are more confortable with. 

Moreover, I don't understand why coordinates could be `Double` or `Long` ? Shouldn't we fix to one unique format ?

Reply:
I personaly prefer the `continue` to explicitly show when we skip and element, and using Kotlins `?:` which are quite convenient here.
Also, it looks like if we store a LatLng with integer values (by example `LatLng(1.0, 2.0)`), firebase will store it as 2 `Long` instead of 2 `Double`. Eventhough this should rarely happen, since a latitude or longitude is way more likely to have decimals, this second cast handles this corner case.

Reply:
I see... Anyway, the `Bin` class that I will create will cast it to a float. For next sprint, please let me now if you find a solution to force the cast on firebase side.